litestar/testing/request_factory.py

```diff
@@ -280,16 +286,7 @@ def _create_request_with_data(
 
         headers = headers or {}
         if data:
-            if isinstance(data, Struct):
-                data = to_builtins(data)
-            elif is_dataclass_instance(data):
-                data = asdict(data)
-            elif is_attrs_class(type(data)):
-                from attr import asdict as attrs_as_dict
-
-                data = attrs_as_dict(data)  # type: ignore[arg-type]
-            elif is_pydantic_model_instance(data):
-                data = data.model_dump(mode="json") if hasattr(data, "model_dump") else json.loads(data.json())
+            data = json.loads(encode_json(data, serializer=get_serializer_from_scope(scope)))
```

Comment:
Do a JSON roundtrip here (because Pydantic) with the regular encoder, so we can ensure this is actually encoded the same as a request, and we don't need special casing for every data type. Incidentally, this make it easier to test.